helios.js

```diff
@@ -26902,212 +26969,217 @@ class DCert extends CborData {
-	 * @param {PubKeyHash} hash
-	 * @param {?Hash} stakingHash
-	 * @returns {Address}
-	 */
-	static fromPubKeyHash(isTestnet, hash, stakingHash = null) {
-		if (stakingHash !== null) {
-			return new Address([isTestnet ? 0x00 : 0x01].concat(hash.bytes).concat(stakingHash.bytes));	
-		} else {
-			return new Address([isTestnet ? 0x60 : 0x61].concat(hash.bytes));
-		}
-	}
-
-	/**
-	 * Simple script address without a staking part
-	 * Only relevant for validator scripts
-	 * @param {boolean} isTestnet
-	 * @param {ValidatorHash} hash
-	 * @param {?Hash} stakingHash
-	 * @returns {Address}
-	 */
-	static fromValidatorHash(isTestnet, hash, stakingHash = null) {
-		if (stakingHash !== null) {
-			return new Address([isTestnet ? 0x10 : 0x11].concat(hash.bytes).concat(stakingHash.bytes));
-		} else {
-			return new Address([isTestnet ? 0x70 : 0x71].concat(hash.bytes));
-		}
-
-	}
-
-	/**
-	 * @returns {string}
-	 */
-	toBech32() {
-		return Crypto.encodeBech32(this.isForTestnet() ? "addr_test" : "addr", this.#bytes);
-	}
-
-	/**
-	 * @returns {Object}
-	 */
-	dump() {
-		return {
-			hex: bytesToHex(this.#bytes),
-			bech32: this.toBech32(),
-		};
-	}
-
-	/**
-	 * @returns {boolean}
-	 */
-	isForTestnet() {
-		let type = this.#bytes[0] & 0b00001111;
-
-		return type == 0;
-	}
-		
-	/**
-	 * @returns {ConstrData}
-	 */
-	toCredentialData() {
-		let vh = this.validatorHash;
-
-		if (vh !== null) {
-			return new ConstrData(1, [
-				new ByteArrayData(vh.bytes)
-			]);
-		} else {
-			let pkh = this.pubKeyHash;
-
-			if (pkh === null) {
-				throw new Error("unexpected");
-			} else {
-				return new ConstrData(0, [
-					new ByteArrayData(pkh.bytes)
-				]);
-			}
-		}
-	}
-
-	/**
-	 * @returns {ConstrData}
-	 */
-	toStakingData() {
-		let sh = this.stakingHash;
-
-		if (sh == null) {
-			return new ConstrData(1, []); // none
-		} else {
-			// some
-			return new ConstrData(0, [
-				// staking credential
-				new ConstrData(0, [
-					// credential (TODO: also allow script and pointer)
-					new ConstrData(0, [
-						new ByteArrayData(sh.bytes),
-					]),
-				])
-			]); // some
-		}
-	}
-
-	/**
-	 * @returns {ConstrData}
-	 */
-	toData() {
-		return new ConstrData(0, [
-			this.toCredentialData(),
-			this.toStakingData(),
-		]);
-	}
-
-	/**
-	 * @type {?PubKeyHash}
-	 */
-	get pubKeyHash() {
-		let type = this.#bytes[0] >> 4;
-
-		if (type % 2 == 0) {
-			return new PubKeyHash(this.#bytes.slice(1, 29));
-		} else {
-			return null;
-		}
-	}
-
-	/**
-	 * @type {?ValidatorHash}
-	 */
-	get validatorHash() {
-		let type = this.#bytes[0] >> 4;
-
-		if (type % 2 == 1) {
-			return new ValidatorHash(this.#bytes.slice(1, 29));
-		} else {
-			return null;
-		}
-	}
-
-	/**
-	 * @type {?Hash}
-	 */
-	get stakingHash() {
-		let type = (this.#bytes[0] >> 4);
-
-		if (type < 4) {
-			let bytes = this.#bytes.slice(29);
-			assert(bytes.length == 28);
-			return new Hash(bytes);
-		} else {
-			return null;
-		}
-	}
+  /** @type {number[]} */
```

Comment:
can you restore the original indentation? now it's difficult to see what exactly changed

Reply:
Bah! Yeah that is my spaces vs tabs set-up! Will fix this up

Reply:
Once we've merged this PR we think of what a good indentation standard could be